lamin_cli/_load.py

```diff
@@ -88,82 +90,95 @@ def script_to_notebook(
+                else:
+                    target_relpath.write_text(transform.source_code)
             else:
-                logger.warning("latest transform run with environment doesn't exist")
+                raise SystemExit("No source code available for this transform.")
+
+            logger.important(f"{transform.type} is here: {target_relpath}")
+
+            if with_env:
+                ln.settings.track_run_inputs = False
+                if (
+                    transform.latest_run is not None
+                    and transform.latest_run.environment is not None
+                ):
+                    filepath_env_cache = transform.latest_run.environment.cache()
+                    target_env_filename = (
+                        target_relpath.parent
+                        / f"{target_relpath.stem}__requirements.txt"
+                    )
+                    shutil.move(filepath_env_cache, target_env_filename)
+                    logger.important(f"environment is here: {target_env_filename}")
+                else:
+                    logger.warning(
+                        "latest transform run with environment doesn't exist"
+                    )
+
+            return target_relpath
+        case "artifact" | "collection":
+            ln.settings.track_run_inputs = False
 
-        return target_relpath
-    elif entity == "artifact":
-        ln.settings.track_run_inputs = False
+            EntityClass = ln.Artifact if entity == "artifact" else ln.Collection
 
-        if query_by_uid:
             # we don't use .get here because DoesNotExist is hard to catch
             # due to private django API
-            artifacts = ln.Artifact.filter(uid__startswith=uid)
-        else:
-            artifacts = ln.Artifact.filter(key=key)
+            if query_by_uid:
+                entities = EntityClass.filter(uid__startswith=uid)
+            else:
+                entities = EntityClass.filter(key=key)
+
+            if (n_entities := len(entities)) == 0:
+                err_msg = f"uid={uid}" if query_by_uid else f"key={key}"
+                raise SystemExit(
+                    f"{entity.capitalize()} with {err_msg} does not exist."
+                )
 
-        if (n_artifacts := len(artifacts)) == 0:
-            err_msg = f"uid={uid}" if query_by_uid else f"key={key}"
-            raise SystemExit(f"Artifact with {err_msg} does not exist.")
+            if n_entities > 1:
+                entities = entities.order_by("-created_at")
 
-        if n_artifacts > 1:
-            artifacts = artifacts.order_by("-created_at")
-        artifact = artifacts.first()
+            entity_obj = entities.first()
+            cache_path = entity_obj.cache()
 
-        cache_path = artifact.cache()
-        logger.important(f"artifact is here: {cache_path}")
-        return cache_path
+            logger.important(f"{entity} is here: {cache_path}")
+            return cache_path
```

Comment:
If it is a collection, then this is a list. Maybe better to format it somehow as a string?

Reply:
Thank you. I missed that because I didn't look at my terminal again after it had finished syncing. Improved stdout.